lib/jobs/local-catalog.js

```diff
@@ -0,0 +1,176 @@
+// Copyright 2015, EMC, Inc.
+/* jshint: node:true */
+
+'use strict';
+
+var di = require('di');
+
+module.exports = LocalCatalogJobFactory;
+di.annotate(LocalCatalogJobFactory, new di.Provide('Job.Local.Catalog'));
+    di.annotate(LocalCatalogJobFactory,
+    new di.Inject(
+        'Job.Base',
+        'JobUtils.CommandParser',
+        'Services.Waterline',
+        'Logger',
+        'Promise',
+        'Assert',
+        'Util',
+        '_',
+        'Services.Encryption',
+        'ChildProcess'
+    )
+);
+function LocalCatalogJobFactory(
+    BaseJob, 
+    parser, 
+    waterline, 
+    Logger, 
+    Promise, 
+    assert, 
+    util, 
+    _,
+    encryption,
+    ChildProcess
+) {
+    var logger = Logger.initialize(LocalCatalogJobFactory);
+
+    /**
+     * @param {Object} options
+     * @param {Object} context
+     * @param {String} taskId
+     * @constructor
+     */
+    function LocalCatalogJob(options, context, taskId) {
+        LocalCatalogJob.super_.call(this, 
+                                   logger,
+                                   options, 
+                                   context, 
+                                   taskId);
+
+        assert.string(this.context.target);
+        assert.arrayOfString(this.options.commands);
+
+        this.nodeId = this.context.target;
+        this.commands = options.commands;
+        this.commands = _.isArray(options.commands) ? options.commands : [options.commands];
+        this.acceptedResponseCodes = options.acceptedResponseCodes[0] || 1;
+    }
+    util.inherits(LocalCatalogJob, BaseJob);
+
+    /**
+     * @memberOf LocalCatalogJob
+     */
+    LocalCatalogJob.prototype._run = function() {
+        var self = this;
+        var cmdResp = [];
+
+        waterline.nodes.findByIdentifier(self.nodeId)
+        .then(function (node) {
+            assert.ok(node, 'No node for local catalog');
+            
+            return Promise.each(self.commands, function(cmd) {
+                return self.runCommand(cmd)
+                .then(function(ret) {
+                    cmdResp.push(ret);
+                });
+            });
+        })
+        .then(function() {
+            return cmdResp;
+        })
+        .then(self.handleResponse.bind(self))
+        .then(function() {
```

Comment:
Since you're doing bind trickery a line above, you can do it here too if you want:

```
.then(self._done.bind(self))
.catch(self._done.bind(self))
```